frontend/src/components/Page.jsx

```diff
@@ -1,71 +1,100 @@
-import React, { useContext, useState } from "react";
+import React, { useContext } from "react";
 
 import Navbar from "./Navbar";
 import userContext from "../contexts/context";
 import Form from "./Form";
-import DishIndex from "./Dish/DishIndex";
+import DishItem from "./Dish/DishItem";
+import BranchItem from "./Branch/BranchItem";
+import BusinessItem from "./Business/BusinessItem";
 
+const _ = require("lodash");
 const fields = require("../fields/index");
-const apiUtil = require("../util/apiUtil");
+const validator = require("../util/validations");
 
-function Page() {
-  const { currentUrl, setUrl, toggleModal, isModalOpen } = useContext(
+function Page({
+  formInfo,
+  sendForm,
+  isAdmin,
+  prepareForm,
+  currentUrl,
+  entitiesState,
+  setEntities,
+  setCurrentUrl,
+}) {
+
+  const { toggleModal, isModalOpen, language } = useContext(
     userContext
   );
-  const [formInfo, setFormInfo] = useState({
-    entityName: "dish",
-    entity: "",
-    postOrPut: "post",
-  });
 
   function dish() {
-    debugger;
-    setUrl("/dish");
+    setCurrentUrl("dish")
+  }
+
+  function branch() {
+    setCurrentUrl("branch")
   }
 
-  function sendForm(formDetails) {
-    debugger
-    if (formInfo.postOrPut === "post") {
-      apiUtil[formInfo.postOrPut](
-        `/${formInfo.entityName}`,
-        formDetails,
-        (data) => {
-          debugger
-          console.log(data);
-        },
-        (err) => {
-          debugger
-          console.log(err);
-        }
-      );
-    } else {
-      apiUtil[formInfo.postOrPut](
-        `/${formInfo.entityName}/${formInfo.entity._id}`,
-        formDetails,
-        (data) => {
-          debugger
-          console.log(data);
-        },
-        (err) => {
-          debugger
-          console.log(err);
-        }
-      );
-    }
+  function business() {
+    setCurrentUrl("business")
   }
 
   return (
     <div>
       <Navbar />
       <button onClick={dish}>Click dish</button>
-      {currentUrl === "/dish" && <DishIndex setFormInfo={setFormInfo} />}
+      <button onClick={branch}>Click branch</button>
+      <button onClick={business}>Click business</button>
+      {isAdmin && !validator.foreignKeysValidator(currentUrl) ? (
+        <div>
+          <button
+            onClick={() => prepareForm("post", {}, currentUrl, {})}
+          >{`Create ${_.capitalize(currentUrl)}`}</button>
+        </div>
+      ) : (
+        ""
+      )}
+      {currentUrl === "dish" &&
+        _.map(entitiesState.dish, (dish) => {
```

Comment:
This param name overrides the function `dish` you defined above, Which reminds me to tell you to make do ESLint test on every PR before you merge. you can add it to the webpack settings to do it automatically

Reply:
What Webpack? to my knowledge - There's no webpack in this project

Reply:
Nvm, just make sure you use ES lint locally on your code once in a while, or before you merge branches. and fix all of it's warnings / errors